.github/workflows/linux.yaml

```diff
@@ -80,3 +100,8 @@ jobs:
           CC: ${{ matrix.compiler }}
         run: |
           test/test_build.sh ubuntu
+      - uses: actions/upload-artifact@v4
+        if: always()
+        with:
+          name: ubuntu-${{ matrix.compiler }}-log-file
```

Comment:
Sorry, I forgot that there were 2 Ubuntu versions !
I suggest `ubuntu-${{ matrix.ubuntu_version }}-${{ matrix.compiler }}-log-file`.

Reply:
You can even drop the initial `ubuntu`, using only `${{ matrix.ubuntu_version }}-${{ matrix.compiler }}-log-file`.